src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/Html/TagBuilder.cs

```diff
@@ -6,41 +6,23 @@
 using System.Globalization;
 using System.IO;
 using System.Text;
+using Microsoft.AspNet.Html.Abstractions;
 using Microsoft.AspNet.Mvc.ViewFeatures;
 using Microsoft.Framework.Internal;
-using Microsoft.Framework.WebEncoders;
 
 namespace Microsoft.AspNet.Mvc.Rendering
 {
     public class TagBuilder
     {
-        private string _innerHtml;
-        private readonly IHtmlEncoder _htmlEncoder;
-
-        public TagBuilder(string tagName)
-            : this(tagName, HtmlEncoder.Default)
-        {
-        }
-
-        public TagBuilder(string tagName, [NotNull] IHtmlEncoder htmlEncoder)
+        public TagBuilder([NotNull] string tagName)
         {
-            if (string.IsNullOrEmpty(tagName))
-            {
-                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, nameof(tagName));
-            }
```

Comment:
Unintended behaviour change. Do not change from checking `IsNullOrEmpty()` to using `[NotNull]`.